### ISSUE TITLE
Incorrect byte swapping with gcc 11.2 and certain compiler options

### ISSUE THREAD

**Issue by emolch** [iris-edu/libmseed#76]
Hi Chad,
thanks a lot for making and maintaining this package. I deeply appreciate your work.

Last week I stumbled over a very surprising problem in a specific combination of gcc version and compiler flags. It is affecting (at least, not sure if only) reading of float64 mseed files. The problem manifests when the compiler aggressively optimizes `ms_gswap8a()`.

To reproduce the problem, try
* gcc 11.2.0 (arch linux ships this as the default since about 2 weeks)
* libmseed 2.19.6
* `CFLAGS="-O3 -flto"  make`
* `make test` should now show `FAILED Test pack-Float64-encoded.test, Compare pack-Float64-encoded.test.ref pack-Float64-encoded.test.out`

The problem does NOT occur when I do either of the following:
* downgrade the compiler, or
* add `-fno-strict-aliasing`, or
* prevent gcc from optimizing `ms_gswap8a` by embedding it in
```
  #pragma GCC push_options
  #pragma GCC optimize ("O0")
  void ms_gswap8a(...) {...}
  #pragma GCC pop_options
  ```
* use `-O1` or remove `-flto`, or
* let the code call `ms_gswap8` instead of `ms_gswap8a`.

 I am not sure what is the right way to fix the problem, what do you think?

Best regards
Sebastian

**Comment by chad-iris** [comment]
Hi @emolch.  Thanks for this report, that's a weird error, impressed you found it.

The byte swapping routines were replaced in PR #75 by @QuLogic.  Can you try the `develop` branch and tell me if the issue is resolved?

thanks in advance!

**Comment by QuLogic** [comment]
This is exactly the problem #74 and #75 are intended to fix.

**Comment by chad-iris** [comment]
I'm going to assume this issue is resolved with the changes in #74.  Please re-open if not the case.
